thonny/codeview.py

```diff
@@ -246,8 +247,8 @@ def set_content_as_bytes(self, data, keep_undo=False):
                 pass
 
             encoding = ask_string(
-                "Bad encoding",
-                "Could not read as %s text.\nYou could try another encoding" % encoding,
+                tr("Bad encoding"),
+                tr("Could not read as {} text.\nYou could try another encoding").format(encoding),
```

Comment:
Most of the translatable strings contain the old style formatting markers, so I suggest you leave `%s` here as well.

Reply:
Yes, I saw it. My decision was based about my experiences. When I teach Python programming and we talk about string formatting in Python, this is something I am not showing to students, or at least I notice them about not to use this style of formatting. I've seen some articles mentioning about not using this style, and also in the official documentation is mentioned, that using such style can be tricky in some situations. Also, when I was translating Mu editor, they were not using "old style" but new one. Question can be about the mentioning the parameter inside of the brackets, so instead of translating string in the form:

```
"Could not read as {} text.\nYou could try another encoding"
```

will be much more "readable" and "explainable" to have string in format:

```
"Could not read as {encoding} text.\nYou could try another encoding"
```

But the final decision is up to you. So I can put it back to "old style".